tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,58 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: SHARED_SECRET
+      default: test-team-snyk
+    - name: ARGS
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.SHARED_SECRET)
+  steps:
+    - name: snyk-code-test
+      image: quay.io/redhat-appstudio/hacbs-test:feature-sast
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
+        if [[ -z $SNYK_TOKEN ]]; then
+          echo "SNYK_TOKEN is empty and a secret 'test-team-snyk' which includes 'snyk_token' need to be created in test-team namespace" > stdout.txt
```

Comment:
Can we create the secret if it fails? This needs to be handled within our tests.

Reply:
We are not sure the user of the task should have the permission to create the secret. So left it as a prework .

Reply:
If anyone tries to run it on their own cluster. It should not fail with SNYK_TOKEN empty.

Reply:
We can't create a secret if it's not available within the user's workspace. The current idea is to have a shared secret available for Red Hat builds, but customers would need to set their their own snyk token. If it's not there, we should report an `ERROR` in the task result (but not fail the task itself of course).

Reply:
I tried in my cluster with latest infra-deployment & created the shared-secret of "test-team-snyk", the task can get the  snyk_token successfully.  It's a shared-secret and not supposed to be created by every user of task.

Reply:
Okay. thanks for the update.